<a href="https://colab.research.google.com/github/Yash-Mahadeshwar/Deep-Learning-with-Keras/blob/master/Lab_8(i)_Keras_metrics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Usage of metrics

A metric is a function that is used to judge the performance of your model. Metric functions are to be supplied in the metrics parameter when a model is compiled.

```
model.compile(loss='mean_squared_error',
              optimizer='sgd',
              metrics=['mae', 'acc'])
from keras import metrics

model.compile(loss='mean_squared_error',
              optimizer='sgd',
              metrics=[metrics.mae, metrics.categorical_accuracy])
```

A metric function is similar to a loss function, except that the results from evaluating a metric are not used when training the model. You may use any of the loss functions as a metric function.

You can either pass the name of an existing metric, or pass a Theano/TensorFlow symbolic function (see Custom metrics).

##Arguments
1. y_true: True labels. Theano/TensorFlow tensor.
2. y_pred: Predictions. Theano/TensorFlow tensor of the same shape as y_true.

##Returns

1. Single tensor value representing the mean of the output array across all datapoints.

#Accuracy
keras.metrics.accuracy(y_true, y_pred)

In [1]:
import keras
from keras import models
from keras.layers import Dense, Dropout
from keras.utils import to_categorical
from keras.datasets import mnist
from keras.utils.vis_utils import model_to_dot
from IPython.display import SVG

NUM_ROWS = 28
NUM_COLS = 28
NUM_CLASSES = 10
BATCH_SIZE = 128
EPOCHS = 10

# Load data
(X_train, y_train), (X_test, y_test) = mnist.load_data()

# Reshape data
X_train = X_train.reshape((X_train.shape[0], NUM_ROWS * NUM_COLS))
X_train = X_train.astype('float32') / 255
X_test = X_test.reshape((X_test.shape[0], NUM_ROWS * NUM_COLS))
X_test = X_test.astype('float32') / 255

# Categorically encode labels
y_train = to_categorical(y_train, NUM_CLASSES)
y_test = to_categorical(y_test, NUM_CLASSES)


# Build neural network
model = models.Sequential()
model.add(Dense(512, activation='elu', input_shape=(NUM_ROWS * NUM_COLS,)))
model.add(Dense(256, activation='elu'))
model.add(Dense(10, activation='softmax'))

# Compile model using accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Using TensorFlow backend.


11493376/11490434 [==============================] - 1s 0us/step





Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where



Train on 60000 samples, validate on 10000 samples
Epoch 1/10





60000/60000 [==============================] - 11s 184us/step - loss: 0.2909 - acc: 0.9122 - val_loss: 0.1547 - val_acc: 0.9524
Epoch 2/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.1202 - acc: 0.9634 - val_loss: 0.1147 - val_acc: 0.9626
Epoch 3/10
60000/60000 [==============================] - 2s 28us/step - loss: 0.0835 - acc: 0.9742 - val_loss: 0.0941 - val_acc: 0.9699
Epoch 4/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0631 - acc: 0.9800 - val_loss: 0.0964 - val_acc: 0.9736
Epoch 5/10
60000/60000 [==============================] - 1s 25us/step - loss: 0.0510 - acc: 0.9836 - val_loss: 0.0795 - val_acc: 0.9774
Epoch 6/10
60000/60000 [==============================] - 2s 25us/step - loss: 0.0395 - acc

#binary_accuracy

keras.metrics.binary_accuracy(y_true, y_pred, threshold=0.5)

In [2]:
# Compile model using binary_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['binary_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.0188 - binary_accuracy: 0.9987 - val_loss: 0.0987 - val_binary_accuracy: 0.9956
Epoch 2/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0159 - binary_accuracy: 0.9990 - val_loss: 0.1109 - val_binary_accuracy: 0.9956
Epoch 3/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0130 - binary_accuracy: 0.9992 - val_loss: 0.1015 - val_binary_accuracy: 0.9960
Epoch 4/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0112 - binary_accuracy: 0.9993 - val_loss: 0.1005 - val_binary_accuracy: 0.9965
Epoch 5/10
60000/60000 [==============================] - 2s 28us/step - loss: 0.0111 - binary_accuracy: 0.9993 - val_loss: 0.1075 - val_binary_accuracy: 0.9960
Epoch 6/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0088 - binary_accuracy: 0.9994 - val_loss: 0.1123 - val_binary_a

#categorical_accuracy

keras.metrics.categorical_accuracy(y_true, y_pred)

In [3]:
# Compile model using categorical_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['categorical_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.0079 - categorical_accuracy: 0.9973 - val_loss: 0.1081 - val_categorical_accuracy: 0.9838
Epoch 2/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0070 - categorical_accuracy: 0.9978 - val_loss: 0.1157 - val_categorical_accuracy: 0.9817
Epoch 3/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0063 - categorical_accuracy: 0.9982 - val_loss: 0.1248 - val_categorical_accuracy: 0.9813
Epoch 4/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.0061 - categorical_accuracy: 0.9980 - val_loss: 0.1372 - val_categorical_accuracy: 0.9803
Epoch 5/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.0060 - categorical_accuracy: 0.9981 - val_loss: 0.1388 - val_categorical_accuracy: 0.9810
Epoch 6/10
60000/60000 [==============================] - 2s 29us/step - loss: 0.0060 - categor

#top_k_categorical_accuracy

keras.metrics.top_k_categorical_accuracy(y_true, y_pred, k=5)

In [4]:
# Compile model using categorical_accuracy
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, y_test))

score = model.evaluate(X_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 30us/step - loss: 0.0042 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1406 - val_top_k_categorical_accuracy: 0.9997
Epoch 2/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0042 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1525 - val_top_k_categorical_accuracy: 0.9999
Epoch 3/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0047 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1516 - val_top_k_categorical_accuracy: 0.9997
Epoch 4/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0041 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1333 - val_top_k_categorical_accuracy: 0.9998
Epoch 5/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.0042 - top_k_categorical_accuracy: 1.0000 - val_loss: 0.1445 - val_top_k_categorical_accuracy: 0.9999
Epoch 6/10
60000/60000 [===========

#sparse_categorical_accuracy

keras.metrics.sparse_categorical_accuracy(y_true, y_pred)

In [5]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['sparse_categorical_accuracy'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.7335 - sparse_categorical_accuracy: 0.5000
Epoch 2/400
 - 0s - loss: 0.7329 - sparse_categorical_accuracy: 0.5000
Epoch 3/400
 - 0s - loss: 0.7324 - sparse_categorical_accuracy: 0.5000
Epoch 4/400
 - 0s - loss: 0.7318 - sparse_categorical_accuracy: 0.5000
Epoch 5/400
 - 0s - loss: 0.7313 - sparse_categorical_accuracy: 0.5000
Epoch 6/400
 - 0s - loss: 0.7308 - sparse_categorical_accuracy: 0.5000
Epoch 7/400
 - 0s - loss: 0.7302 - sparse_categorical_accuracy: 0.5000
Epoch 8/400
 - 0s - loss: 0.7297 - sparse_categorical_accuracy: 0.5000
Epoch 9/400
 - 0s - loss: 0.7291 - sparse_categorical_accuracy: 0.5000
Epoch 10/400
 - 0s - loss: 0.7286 - sparse_categorical_accuracy: 0.5000
Epoch 11/400
 - 0s - loss: 0.7281 - sparse_categorical_accuracy: 0.5000
Epoch 12/400
 - 0s - loss: 0.7276 - sparse_categorical_accuracy: 0.5000
Epoch 13/400
 - 0s - loss: 0.7271 - sparse_categorical_accuracy: 0.5000
Epoch 14/400
 - 0s - loss: 0.7266 - sparse_categorical_accuracy: 0.5000
E

#sparse_top_k_categorical_accuracy

keras.metrics.sparse_top_k_categorical_accuracy(y_true, y_pred, k=5)

In [6]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['sparse_top_k_categorical_accuracy'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.6384 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 2/400
 - 0s - loss: 0.6380 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 3/400
 - 0s - loss: 0.6377 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 4/400
 - 0s - loss: 0.6373 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 5/400
 - 0s - loss: 0.6370 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 6/400
 - 0s - loss: 0.6366 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 7/400
 - 0s - loss: 0.6363 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 8/400
 - 0s - loss: 0.6359 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 9/400
 - 0s - loss: 0.6356 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 10/400
 - 0s - loss: 0.6352 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 11/400
 - 0s - loss: 0.6349 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 12/400
 - 0s - loss: 0.6345 - sparse_top_k_categorical_accuracy: 0.5000
Epoch 13/400
 - 0s - loss: 0.6342 - sparse_top_k_categorical_accuracy: 0.

#cosine_proximity

keras.metrics.cosine_proximity(y_true, y_pred, axis=-1)

In [7]:
from numpy import array
from keras.models import Sequential
from keras.layers import Dense
from matplotlib import pyplot
# prepare sequence
X = array([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0])
y = array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1])
# create model
model = Sequential()
model.add(Dense(2, input_dim=1))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['cosine_proximity'])
# train model
history = model.fit(X, y, epochs=400, batch_size=len(X), verbose=2)

Epoch 1/400
 - 0s - loss: 0.8159 - cosine_proximity: -5.0000e-01
Epoch 2/400
 - 0s - loss: 0.8153 - cosine_proximity: -5.0000e-01
Epoch 3/400
 - 0s - loss: 0.8147 - cosine_proximity: -5.0000e-01
Epoch 4/400
 - 0s - loss: 0.8139 - cosine_proximity: -5.0000e-01
Epoch 5/400
 - 0s - loss: 0.8132 - cosine_proximity: -5.0000e-01
Epoch 6/400
 - 0s - loss: 0.8124 - cosine_proximity: -5.0000e-01
Epoch 7/400
 - 0s - loss: 0.8117 - cosine_proximity: -5.0000e-01
Epoch 8/400
 - 0s - loss: 0.8109 - cosine_proximity: -5.0000e-01
Epoch 9/400
 - 0s - loss: 0.8102 - cosine_proximity: -5.0000e-01
Epoch 10/400
 - 0s - loss: 0.8094 - cosine_proximity: -5.0000e-01
Epoch 11/400
 - 0s - loss: 0.8087 - cosine_proximity: -5.0000e-01
Epoch 12/400
 - 0s - loss: 0.8079 - cosine_proximity: -5.0000e-01
Epoch 13/400
 - 0s - loss: 0.8072 - cosine_proximity: -5.0000e-01
Epoch 14/400
 - 0s - loss: 0.8064 - cosine_proximity: -5.0000e-01
Epoch 15/400
 - 0s - loss: 0.8057 - cosine_proximity: -5.0000e-01
Epoch 16/400
 - 0s 

#Thank you for completing the notebook

*ACCURACY SCORES*

**Accuracy-**

Test loss: 0.10376196658944027


Test accuracy: 0.9757


**binary_accuracy-**

Test loss: 0.15646206490577766


Test accuracy: 0.9955399974822998


**categorical_accuracy-**

Test loss: 0.14195446051103405


Test accuracy: 0.9816


**top_k_categorical_accuracy-**

Test loss: 0.15224556388646485


Test accuracy: 0.9999


**sparse_categorical_accuracy-**

loss: 0.4375 


sparse_categorical_accuracy: 0.5000


**sparse_top_k_categorical_accuracy-**

loss: 0.7547 


sparse_top_k_categorical_accuracy: 0.5000


**cosine_proximity-**

loss: 0.7574 


cosine_proximity: -5.0000e-01

The best metric among all is **top_k_categorical_accuracy** which gives almost an accuracy of 1.

In [8]:
from keras.datasets import cifar10
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation, Flatten
from keras.utils import to_categorical

#import dataset
(X_train, Y_train), (X_test, Y_test) = cifar10.load_data()

#change shape from image to vector
X_train = X_train.reshape(50000, 32 * 32 * 3)
X_test = X_test.reshape(10000, 32 * 32 * 3)

#preprocess
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')
X_train /= 255.0
X_test /= 255.0

#change labels from numeric to one hot encoded
Y_train = to_categorical(Y_train, 10)
Y_test =  to_categorical(Y_test, 10)

#Model building
model = Sequential()
model.add(Dense(1024, input_shape=(3072, )))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(10))
model.add(Activation('softmax'))

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

170500096/170498071 [==============================] - 4s 0us/step
Train on 50000 samples, validate on 10000 samples
Epoch 1/10
50000/50000 [==============================] - 3s 51us/step - loss: 3.5652 - top_k_categorical_accuracy: 0.7461 - val_loss: 1.8930 - val_top_k_categorical_accuracy: 0.8295
Epoch 2/10
50000/50000 [==============================] - 2s 42us/step - loss: 1.8093 - top_k_categorical_accuracy: 0.8459 - val_loss: 1.8034 - val_top_k_categorical_accuracy: 0.8424
Epoch 3/10
50000/50000 [==============================] - 2s 41us/step - loss: 1.7030 - top_k_categorical_accuracy: 0.8738 - val_loss: 1.6407 - val_top_k_categorical_accuracy: 0.8839
Epoch 4/10
50000/50000 [==============================] - 2s 42us/step - loss: 1.6297 - top_k_categorical_accuracy: 0.8877 - val_loss: 1.7357 - val_top_k_categorical_accuracy: 0.8706
Epoch 5/10
50000/50000 [==============================] - 2s 41us/step - loss: 1.5776 - top_k_categorical_accuracy: 0.8974 - val_loss: 1.5338 - val_top

**CIFAR 10**

Test loss: 1.4908046955108643


Test accuracy: 0.914

In [9]:
from keras.datasets import cifar100
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation, Flatten
from keras.utils import to_categorical

#import dataset
(X_train, Y_train), (X_test, Y_test) = cifar100.load_data()

#change shape from image to vector
X_train = X_train.reshape(50000, 32 * 32 * 3)
X_test = X_test.reshape(10000, 32 * 32 * 3)

#preprocess
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')
X_train /= 255.0
X_test /= 255.0

#change labels from numeric to one hot encoded
Y_train = to_categorical(Y_train, 100)
Y_test =  to_categorical(Y_test, 100)

#Model building
model = Sequential()
model.add(Dense(1024, input_shape=(3072, )))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dense(100))
model.add(Activation('softmax'))


from keras import optimizers
adamax=keras.optimizers.Adamax(lr=0.002, beta_1=0.9, beta_2=0.999)

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

169009152/169001437 [==============================] - 4s 0us/step
Train on 50000 samples, validate on 10000 samples
Epoch 1/10
50000/50000 [==============================] - 3s 54us/step - loss: 4.2813 - top_k_categorical_accuracy: 0.2046 - val_loss: 3.9339 - val_top_k_categorical_accuracy: 0.2902
Epoch 2/10
50000/50000 [==============================] - 2s 42us/step - loss: 3.7835 - top_k_categorical_accuracy: 0.3391 - val_loss: 3.6975 - val_top_k_categorical_accuracy: 0.3692
Epoch 3/10
50000/50000 [==============================] - 2s 40us/step - loss: 3.5984 - top_k_categorical_accuracy: 0.3944 - val_loss: 3.8066 - val_top_k_categorical_accuracy: 0.3652
Epoch 4/10
50000/50000 [==============================] - 2s 44us/step - loss: 3.4606 - top_k_categorical_accuracy: 0.4330 - val_loss: 3.4495 - val_top_k_categorical_accuracy: 0.4472
Epoch 5/10
50000/50000 [==============================] - 2s 41us/step - loss: 3.3541 - top_k_categorical_accuracy: 0.4608 - val_loss: 3.5586 - val_top

**CIFAR 100**

Test loss: 3.393477967834473


Test accuracy: 0.483

In [10]:
import keras
from keras import models
from keras.layers import Dense, Dropout
from keras.utils import to_categorical
from keras.datasets import fashion_mnist
from keras.utils.vis_utils import model_to_dot
from IPython.display import SVG

NUM_ROWS = 28
NUM_COLS = 28
NUM_CLASSES = 10
BATCH_SIZE = 128
EPOCHS = 10

# Load data
(X_train, Y_train), (X_test, Y_test) = fashion_mnist.load_data()



# Reshape data
X_train = X_train.reshape((X_train.shape[0], NUM_ROWS * NUM_COLS))
X_train = X_train.astype('float32') / 255
X_test = X_test.reshape((X_test.shape[0], NUM_ROWS * NUM_COLS))
X_test = X_test.astype('float32') / 255

# Categorically encode labels
Y_train = to_categorical(Y_train, NUM_CLASSES)
Y_test = to_categorical(Y_test, NUM_CLASSES)


# Build neural network
model = models.Sequential()
model.add(Dense(512, activation='relu', input_shape=(NUM_ROWS * NUM_COLS,)))
model.add(Dense(256, activation='relu'))
model.add(Dense(10, activation='softmax'))

# Compile model using above optimizer
model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['top_k_categorical_accuracy'])

# Train model
model.fit(X_train, Y_train,
          batch_size=BATCH_SIZE,
          epochs=EPOCHS,
          verbose=1,
          validation_data=(X_test, Y_test))

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

4423680/4422102 [==============================] - 1s 0us/step
Train on 60000 samples, validate on 10000 samples
Epoch 1/10
60000/60000 [==============================] - 2s 35us/step - loss: 0.5412 - top_k_categorical_accuracy: 0.9932 - val_loss: 0.4395 - val_top_k_categorical_accuracy: 0.9971
Epoch 2/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.3751 - top_k_categorical_accuracy: 0.9978 - val_loss: 0.3806 - val_top_k_categorical_accuracy: 0.9973
Epoch 3/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.3337 - top_k_categorical_accuracy: 0.9983 - val_loss: 0.3895 - val_top_k_categorical_accuracy: 0.9972
Epoch 4/10
60000/60000 [==============================] - 2s 26us/step - loss: 0.3094 - top_k_categorical_accuracy: 0.9988 - val_loss: 0.3517 - val_top_k_categorical_accuracy: 0.9978
Epoch 5/10
60000/60000 [==============================] - 2s 27us/step - loss: 0.2913 - top_k_categorical_accuracy: 0.9989 - val_loss: 0.3425 - val_top_k_c

**FASHION_MNIST**

Test loss: 0.34572943717837334


Test accuracy: 0.9981